### DISCUSSION TITLE
The given version is not supported

### DISCUSSION THREAD

**Discussion by metalglove** [microsoft/onnxruntime#9630] [Other Q&A]
I am trying to create a C++ inference app for my model.

I have cloned the onnxruntime with tag v1.9.1 recursively and build the source code on Windows with this command using the Visual Studio 2019 Developer Terminal:
```
build.bat --config Release --skip_tests --build_shared_lib --parallel
```
Using CMake I have linked the library:
```
set(ONNXRUNTIME_ROOTDIR "${CMAKE_SOURCE_DIR}/onnxruntime")
include_directories("${ONNXRUNTIME_ROOTDIR}/include")
find_library(ONNXRUNTIME_LIBRARY onnxruntime HINTS "${ONNXRUNTIME_ROOTDIR}/lib")

add_executable(onnx-inference inference.cpp)
target_link_libraries(onnx-inference PRIVATE ${ONNXRUNTIME_LIBRARY})
```
Upon importing the `onnxruntime_cxx_api.h` I receive this message in the output:
```
The given version [9] is not supported, only version 1 to 7 is supported in this build.
Hello, world!
```
Have I done something wrong?

This is my code:
```C++
#include <iostream>
#include "onnxruntime/core/session/onnxruntime_cxx_api.h"

int main(int, char**) 
{
    std::cout << "Hello, world!\n";
}
```

**Comment by snnn** [comment]
We need to know which onnxruntime.dll was used. 

To identify the problem:
1. Install [windbg](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools)
2.  Find the gflags.exe in the windbg install folder and start it, 

3. Select the image file tab
![image](https://user-images.githubusercontent.com/856316/74082914-91d96900-4a13-11ea-95eb-d73db8a5cf13.png)

4. Type your exe name then press the "tab" key, check "Show loader snaps" box

![image](https://user-images.githubusercontent.com/856316/74082958-b7667280-4a13-11ea-9eaf-6752695b4904.png)


5.  Start windbg, run your application 
![image](https://user-images.githubusercontent.com/856316/74083043-7d49a080-4a14-11ea-90fa-2960544a3dce.png)

7. in the command windows, type "g", then press the enter key
![image](https://user-images.githubusercontent.com/856316/74083078-d285b200-4a14-11ea-9df4-e86c421c5617.png)

7. You'll see a lot of outputs in the debugging windows.  Please search the lines that contains "onnxruntime.dll" and post them here.  You should be able see a full path of onnxruntime.dll. Does it live in C:\windows\system32 or your application dir? Once you find the DLL, please right click the dll and find out the version of it in its properties page.


Please post the log here if you can't find the problematic DLL by yourself.

**Reply by metalglove** [reply]
I will try this out tonight and report back.

**Comment by snnn** [comment]
And I guess you were using Windows 11.

**Reply by metalglove** [reply]
That is correct.

**Reply by snnn** [reply]
Don't rely on PATH env to locate DLLs.  Instead, please put onnxruntime.dll within the same folder as your main exe.

**Reply by metalglove** [reply]
That is what I did. I have a repository to replicate it exactly. The CMakeLists puts it in the build dir with the exe. 
https://github.com/metalglove/inference-in-cpp

**Reply by metalglove** [reply]
https://github.com/metalglove/inference-in-cpp/blob/0a6754a897de589442318d13dc4c4603a42fd52d/src/onnx/CMakeLists.txt#L6

Change this line to use v1.9.1.
